### PR TITLE
By default, disabled students should be filtered, with an optional override

### PR DESCRIPTION
**Pull request by jmealo**
https://www.wrike.com/open.htm?id=44109172

### CODE REVIEW

File: php-classes/Slate/People/Student.php [hunk 1]
```diff
@@ -104,14 +104,16 @@ public static function getDistinctGraduationYears()
...
+    public static function getAllByListIdentifier($identifier, $includeDisabled = false)
     {
         if (!$identifier) {
             throw new \Exception('');
         }
 
         if ($identifier == 'all') {
-            return static::getAllByClass();
+            return array_filter(static::getAllByClass(), function ($Person) use ($includeDisabled) {
+                return ($isDisabled || $Person->AccountLevel != 'Disabled');
```
Comment (themightychris) [inline php-classes/Slate/People/Student.php]: s/isDisabled/includeDisabled/

File: php-classes/Slate/People/Student.php [hunk 2]
```diff
@@ -122,16 +124,16 @@ public static function getAllByListIdentifier($identifier)
                     throw new \Exception('Group not found');
                 }
 
-                return array_filter($Group->getAllPeople(), function($Person) {
-                    return $Person->isA(Student::class);
+                return array_filter($Group->getAllPeople(), function($Person) use ($includeDisabled) {
+                    return $Person->isA(Student::class) && ($isDisabled || $Person->AccountLevel != 'Disabled');
```
Comment (themightychris) [inline php-classes/Slate/People/Student.php]: s/isDisabled/includeDisabled/

File: php-classes/Slate/People/Student.php [hunk 3]
```diff
@@ -122,16 +124,16 @@ public static function getAllByListIdentifier($identifier)
...
                 });
             case 'section':
                 if (!$Section = Section::getByHandle($groupHandle)) {
                     throw new \Exception('Section not found');
                 }
 
-                return array_values(array_filter($Section->Students, function($Person) {
-                    return $Person->isA(Student::class);
+                return array_values(array_filter($Section->Students, function($Person) use ($includeDisabled) {
+                    return $Person->isA(Student::class) && ($isDisabled || $Person->AccountLevel != 'Disabled');
```
Comment (themightychris) [inline php-classes/Slate/People/Student.php]: s/isDisabled/includeDisabled/

**Comment by jmealo** [comment]
@themightychris good catch, I screwed that up when I switched repos. Fixing now.

**Pull request merged**